pkg/processor/cascadingfilterprocessor/sampling/drop_trace_factory.go

```diff
@@ -28,10 +30,31 @@ type dropTraceEvaluator struct {
 	stringAttr  *stringAttributeFilter
 	attrs       []attributeFilter
 	operationRe *regexp.Regexp
+	statusCode  *string
 
 	logger *zap.Logger
 }
 
+func validateStatusCode(statusCode *string) error {
+	if statusCode == nil {
+		return nil
+	}
+
+	validStatusCodes := []string{
+		ptrace.StatusCodeError.String(),
+		ptrace.StatusCodeOk.String(),
+		ptrace.StatusCodeUnset.String(),
+	}
+
+	for _, valid := range validStatusCodes {
+		if *statusCode == valid {
+			return nil
+		}
+	}
+
+	return errors.New("invalid status code: must be one of 'Error', 'Ok', or 'Unset' ")
```

Comment:
```suggestion
	return errors.New("invalid status code: must be one of 'Error', 'Ok', or 'Unset'")
```

I was talking about above string 😅
It's up to you. It can go that way 🤗